src/main/java/org/webshar/hrms/model/db/Employee.java

```diff
@@ -41,14 +41,16 @@
   private String email;
   @Column(name = "joining_date", nullable = false, updatable = false)
   private Date joiningDate;
-  @Column(name = "exit_date", nullable = true)
+  @Column(name = "exit_date")
   private Date exitDate;
-  @Column(name = "address", nullable = true, length = 300)
+  @Column(name = "address", length = 300)
   private String address;
   @Column(name = "contact", nullable = false, length = 100)
   private String contact;
   @Column(name = "designation", nullable = false, length = 200)
   private String designation;
+  @Column(name = "reports_to", length = 11)
+  private Long reportsTo;
```

Comment:
- `private Long reportsTo;` This does not correctly model/enforce object type relationship "employee reports to another employee". Due to this, there is a risk of developers doing errors by populating any long value in this field unknowingly. 
- This needs to be improved by specifying the correct type as follows: `@OneToOne private Employee reportsTo;` 
- Most of the time, from the UI display point of view we need to show employee name to whom one is reporting to and not `id`.  This can be achieved by a single read operation managed by JPA if the `Employee` type is specified. Otherwise, if a `Long` type is used, then one more manual read operation will be needed to get the details about `reportsTo` employee.

Reply:
Agreed and made & pushed the necessary change please check

Reply:
thanks